### PR TITLE
fixed version checks for force logout

### PR DESCRIPTION
**Pull request by paynejacob**
https://github.com/rancher/rancher/issues/34149

Fixed how we decide to force a logout. All users will be logged out in ONLY the following cases:
- if the current rancher version is an rc or release and the target version is greater than the last force logout
- if the current rancher version is a `*-head` build and the last force logout version does not match the current version

### CODE REVIEW

File: pkg/rancher/migrations.go
Comment (snasovich) [inline pkg/rancher/migrations.go:48]:
```suggestion
	// if no last migration is found we always run force logout
```

File: pkg/rancher/migrations.go
Comment (snasovich) [inline pkg/rancher/migrations.go:52]: Could still use `<=` here. If versions are the same, we don't migrate. No need to leave it to the next step.

**Review by snasovich — APPROVED** [review]
Couple nits, otherwise LGTM

**Review by snasovich — APPROVED** [review]
LGTM, still need committer review

**Review by kinarashah — COMMENTED** [review]
checks based on rancher version look good. I have a question on the logout flow itself tho, we seem to be deleting tokens first followed by storing the rancher version in the configmap. If the update errors out, we'd delete the tokens again, is that expected?

**Comment by paynejacob** [comment]
@kinarashah yes that would be the case.  We had a previous pr to add logic to ensure that didn't happen but it was determined the logic was too complex for such an uncommon situation.  if rancher crashes during this part, it is crashing during the first startup after an upgrade and is likely in an unusable state anyway.

**Comment by kinarashah** [comment]
I was thinking along the lines of error like an update conflict, but in that case we'd have seen the logs twice. I don't see that in the linked issue so looks good for now.

**Review by kinarashah — APPROVED** [review]

File: go.mod
Comment (dramich) [inline go.mod:137]: This is pulling in a 4th semver package... can an existing be used?

Comment (dramich) [reply go.mod:137]: Or we open an issue to get them consolidated

Comment (paynejacob) [reply go.mod:137]: I would argue we should consolidate, looking at the other ones in here they are all the same.  I can reuse one but I will have to write a wrapper func to validate them.

Comment (dramich) [reply go.mod:137]: Going with the official package seems like the best option anyways.

**Review by dramich — APPROVED** [review]

**Pull request merged**
